---
title: null
published: false
subtitle: null
layout: page
---


***

## Angel Angelov
[NGS Competence Center Tübingen](https://portal.qbic.uni-tuebingen.de/portal/web/ncct/ncct_welcome)   
angel.angelov[at]med.uni-tuebingen.de

***

Education
----------------

**2001-2004** \| **PhD, Microbiology and Bioinformatics**; [Georg-August-University Göttingen](https://www.uni-goettingen.de/)

*Thesis title:* Genome sequence analysis and characterization of recombinant enzymes from the
thermoacidophilic archaeon *Picrophilus torridus*

**1993-1999** \| **MSc, Molecular biology and Genetics**; [St. Kliment Ohridski University](https://www.uni-sofia.bg/eng) (Sofia, Bulgaria)

***

Research interests
--------------------------
  * microbial genomics, metagenomics and genetics
  * NGS technologies, bioinformatics
  * metabolic and enzyme engineering, carbohydrate-active enzymes
  * natural transformation

***

Experience
-------------

**Since 2019** \| **Scientist**, [NGS Competence Center Tübingen](https://portal.qbic.uni-tuebingen.de/portal/web/ncct/ncct_welcome), Institute of Medical Microbiology and Hygiene, University Hospital Tübingen

* Bioinformatics pipeline development
* Lab and project management, consulting
* Lab automation


**2009-2019** \| **Group leader**, [Department of Microbiology](http://mibio.wzw.tum.de/index.php?id=4), Technical University of Munich, Germany

* Development of new tools for metagenomic library construction
* Screening, purification and biochemical characterization of polysaccharide-degrading enzymes from extremophilic microorganisms and from metagenomic libraries
* Natural transformation in Gram-positive microorganisms
* Metabolic engineering of *Thermus thermophilus* and *Micrococcus luteus*

**2004-2009** \| **Postdoctoral fellow**, [Institute of Microbiology and Genetics](https://www.uni-goettingen.de/en/11845.html), Georg-August-University Göttingen

* New cellulolytic and hemicellulolytic enzymes derived from the metagenomes of cellulosic biomass-degrading microbial consortia

**2001-2004** \| **Research assistant**,  Institute of Microbiology and Genetics, Georg-August-University Göttingen

* Genome sequence analysis and characterization of recombinant enzymes from the thermoacidophilic archaeon *Picrophilus torridus*

**2000-2001** \| **Research assistant**, [National Center of Infectious and Parasitic Diseases](http://www.ncipd.org), Sofia, Bulgaria

* Linking HIV genotype variation and antiviral therapy outcome

**1998-1999** \| **Student research assistant**, [University Obstetrics and Gynecology Hospital](http://maichindom.com/en/)

* Investigation of the frequency of a genetically determined resistance to HIV infection in the Bulgarian population

***

Fields of expertise
--------------------

**Programming and data analysis**  
Advanced NGS and proteomics data analysis, focus on metagenomics, genome sequence assembly and annotation, differential gene expression. Bioinformatics pipeline development. Expertise in R, Shiny, shell scripting, aws, git, docker, nextflow.

**R programming language**  
Exploratory data analysis, visualization and modelling. Shiny app development

* [GitHub repository](https://github.com/angelovangel)
* [R-blog](https://angelovangel.github.io/)
* examples of Shiny apps written by me, [running in the cloud](https://angelovangel.github.io/portfolio)

**General and applied microbiology, genetics**  
Cultivation of extremophilic microorganisms, metabolic engineering, development of genetic tools, natural transformation, type IV pili

**Biochemistry**  
Protein production, purification and analysis, carbohydrate-active enzymes, carbohydrate-binding modules

**Analytical tecniques**  
Extensive experience in NGS, proteomics, chromatography and spectrometry techniques (FPLC, HPLC, GC-MS), automated liquid handling systems



***

Language skills
----------------
Bulgarian - native  
English - fluent  
German - fluent  

***

Peer review and editing
----------------------
**Since 2013** \| Assistant Editor, [Frontiers in Systems Microbiology](https://www.frontiersin.org/journals/microbiology/sections/systems-microbiology) (28 editorial contributions)  
**Since 2012** \| Reviewer, [American Journal Experts](https://www.aje.com)   
**Since 2013** \| Reviewer, [Frontiers in Microbiology](https://www.frontiersin.org/journals/microbiology)   

Numerous _ad hoc reviews_ for the journals Applied and Environmental Microbiology, Journal of Bacteriology, Extremophiles, Biotechnology for Biofuels, BMC Genomics, Systematic and Applied Microbiology, Scientific Reports and other journals.

***

Selected publications
-----------------

*for a complete list see my [google scholar](https://scholar.google.com/citations?user=J7YO6I4AAAAJ) or [pubmed](https://www.ncbi.nlm.nih.gov/myncbi/angel.angelov.1/bibliography/public/) profile*

Jarzab A, Kurzawa N, Hopf T, Moerch M, Zecha J, Leijten N, Bian Y, Musiol E, Maschberger M, Stoehr G, Becher I, Daly C, Samaras P, Mergner J, Spanier B, <u>Angelov A</u>, Werner T, Bantscheff M, Wilhelm M, Klingenspor M, Lemeer S, Liebl W, Hahne H, Savitski MM, Kuster B.    
[Meltome atlas-thermal proteome stability across the tree of life](https://www.nature.com/articles/s41592-020-0801-4). **Nat Methods.** 2020 May;17(5):495-503

Surger M, <u>Angelov A</u>, Liebl W.   
[Distribution and diversity of olefins and olefin-biosynthesis genes in Gram-positive bacteria](https://pubmed.ncbi.nlm.nih.gov/32313552/)   
**Biotechnol Biofuels.** 2020;13:70.

Lichev A, <u>Angelov A</u>, Cucurull I, Liebl W.   
[Amino acids as nutritional factors and (p)ppGpp as an alarmone of the stringent response regulate natural transformation in _Micrococcus luteus_](https://pubmed.ncbi.nlm.nih.gov/31363120/)    
**Sci Rep.** 2019 Jul 30;9(1):11030


<u>Angelov A</u>, Übelacker M, Liebl W.    
[Variances in cellular sedimentation behavior as an effective enrichment method of hydrocarbon-overproducing *Micrococcus luteus* strains](https://www.ncbi.nlm.nih.gov/pubmed/30377447)   
**Biotechnol Biofuels**. 2018 ;11:288

<u>Angelov A</u>, Pham VTT, Übelacker M, Brady S, Leis B, Pill N, Brolle J, Mechelke M, Moerch M, Henrissat B, Liebl W.   
[A metagenome-derived thermostable β-glucanase with an unusual module architecture which defines the new glycoside hydrolase family GH148](https://www.ncbi.nlm.nih.gov/pubmed/29229913)    
**Sci Rep**. 2017; 7(1):17306

Schiefner A, <u>Angelov A</u>, Liebl W, Skerra A.    
[Structural basis for cellulose binding by the type A carbohydrate-binding module 64 of *Spirochaeta thermophila*](https://www.ncbi.nlm.nih.gov/pubmed/26868291).   
**Proteins**. 2016; 84(6):855-8

<u>Angelov A</u>, Bergen P, Nadler F, Hornburg P, Lichev A, Übelacker M, Pachl F, Kuster B, Liebl W.   
[Novel Flp pilus biogenesis-dependent natural transformation](https://www.ncbi.nlm.nih.gov/pubmed/25713572).   
**Frontiers in microbiology**. 2015; 6:84

<u>Angelov A</u>, Li H, Geissler A, Leis B, Liebl W.   
[Toxicity of indoxyl derivative accumulation in bacteria and its use as a new counterselection principle](https://www.ncbi.nlm.nih.gov/pubmed/23871391).   
**Systematic and applied microbiology**. 2013; 36(8):585-92

<u>Angelov A</u>, Loderer C, Pompei S, Liebl W.   
[Novel family of carbohydrate-binding modules revealed by the genome sequence of _Spirochaeta thermophila_ DSM 6192](https://www.ncbi.nlm.nih.gov/pubmed/21685171).    
**Applied and environmental microbiology**. 2011; 77(15):5483-9

<u>Angelov A</u>, Mientus M, Liebl S, Liebl W.    
[A two-host fosmid system for functional screening of (meta)genomic libraries from extreme thermophiles](https://www.ncbi.nlm.nih.gov/pubmed/19285378).   
**Systematic and applied microbiology**. 2009; 32(3):177-85

Fütterer O, <u>Angelov A</u>, Liesegang H, Gottschalk G, Schleper C, Schepers B, Dock C, Antranikian G, Liebl W.   
[Genome sequence of Picrophilus torridus and its implications for life around pH 0](https://www.ncbi.nlm.nih.gov/pubmed/15184674).   
**PNAS**. 2004; 101(24):9091-6
